install/CMakeLists.txt

```diff
@@ -35,29 +37,44 @@ add_custom_target(BUILD_version ALL
         BYPRODUCTS "${CMAKE_CURRENT_BINARY_DIR}/VERSION")
 
 
-install(FILES
-    "${CMAKE_CURRENT_BINARY_DIR}/VERSION"
+function(install_config_files FILES DEST_DIR)
+    foreach(FILE_NAME ${FILES})
+        get_filename_component(FILE_BASE ${FILE_NAME} NAME)
+        if (NOT EXISTS "${DEST_DIR}/${FILE_BASE}" OR FORCE_CONF_OVERWRITE)
+            message(STATUS "Installing ${FILE_BASE}")
+            install(FILES ${FILE_NAME}
+                    COMPONENT common
+                    DESTINATION ${DEST_DIR})
+        else()
+            message(STATUS "${FILE_BASE} already exists. Not overwriting, use the FORCE_CONF_OVERWRITE option to overwrite.")
+        endif()
+    endforeach()
+endfunction()
+
+set(FO_CONFIG_FILES
     ${CMAKE_CURRENT_BINARY_DIR}/gen/Db.conf
     ${CMAKE_CURRENT_BINARY_DIR}/gen/fossology.conf
+)
+install_config_files("${FO_CONFIG_FILES}" "${FO_SYSCONFDIR}")
+
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/VERSION"
+    DESTINATION ${FO_SYSCONFDIR}
     COMPONENT common
-    DESTINATION ${FO_SYSCONFDIR})
+)
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/VERSION"
     DESTINATION ${FO_MODDIR}
     COMPONENT common
 )
 
-
-install(DIRECTORY
-    ${CMAKE_CURRENT_SOURCE_DIR}/defconf/
-    ${CMAKE_CURRENT_SOURCE_DIR}/fossdash/
-    COMPONENT common
-    DESTINATION ${FO_SYSCONFDIR}
-    FILES_MATCHING
-    PATTERN *.txt
-    PATTERN *.yml
+set(FO_DIR_FILES
+    ${CMAKE_CURRENT_SOURCE_DIR}/defconf/samplefooter.txt
```

Comment:
Would recommend using `file(GLOB)` function to retain the existing functionality to copy all `.txt` and `.yml` files from respective directories.
This will help adding more configurations files in future easy.

Check documentation at: https://cmake.org/cmake/help/v3.13/command/file.html#glob

Reply:
The first one looks good.